pypeit/scripts/ql_keck_deimos.py

```diff
@@ -0,0 +1,239 @@
+        import argparse
+        parser = super().get_parser(description='Pypeit QL for Keck/DEIMOS',
+                                    width=width, formatter=argparse.RawDescriptionHelpFormatter)
+        parser = argparse.ArgumentParser(description='Script to run PypeIt in QuickLook on a set of '
+                                                    'Keck/DEIMOS files',
+                                        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+        parser.add_argument('full_rawpath', type=str, help='Full path to the raw files')
+        parser.add_argument('--redux_path', type=str, 
+                            help='Path to where reduction products lie')
+        parser.add_argument('--root', type=str, help='Root of filenames, eg.  DE.2018')
+        parser.add_argument('--calibs_only', default=False, action='store_true',
+                            help='Run on calibs only')
+        parser.add_argument('--science', type=str, help='Science frame filename')
+        parser.add_argument('-b', '--box_radius', type=float,
+                            help='Set the radius for the boxcar extraction (arcsec)')
+        parser.add_argument('-d', '--det', type=int, default=0,
+                            help='Detector number. Cannot use with --slit_spat')
+        parser.add_argument('--ignore_headers', default=False, action='store_true',
+                            help='Ignore bad headers?')
+        parser.add_argument('--user_pixflat', type=str,
+                            help='Use a user-supplied pixel flat (e.g. keck_lris_blue)')
+        parser.add_argument('--maskID', type=int,
+                            help='Reduce this slit as specified by the maskID value')
+        parser.add_argument('--slit_spat', type=str,
+                            help='Reduce only this slit on this detector DET:SPAT_ID, e.g. 0:175')
+
+        return parser
+
+    @staticmethod
+    def main(pargs):
+
+        script_Utils = Utilities('keck_deimos')
+
+        # Afternoon calibs
+        if pargs.calibs_only:
+            if pargs.root is None:
+                msgs.error("Need to set --root when using --calibs_only")
+            msgs.info("Processing calibrations only")
+            process_calibs(pargs, script_Utils)
+            return
+
+        # Slurp the afternoon runs and grab the right PypeIt file
+        calib_pypeit_file, ps_sci = get_science_setup(pargs, script_Utils)
+
+        # Run on the science frame
+        run_on_science(pargs, script_Utils, calib_pypeit_file, ps_sci)
+
+
+def process_calibs(pargs, script_Utils):
+    """
+    Process the calibration files
+
+    Args:
+        pargs (argparse.ArgumentParser):
+            Command line arguments
+        script_Utils (:class:`pypeit.scripts.utilts.Utilities`):
+    """
+    # Slurp
+    output_path = pargs.redux_path
+
+    # Run setup
+    ps, setups, indx = script_Utils.run_setup(os.path.join(pargs.full_rawpath, pargs.root),
```

Comment:
I have noticed that still the script tends to reduce not only the science frame that is specified in the call of the script but also other science files that for some reason ended up in the `deimos_calib.pypeit` file . So I think it's really important that here the science files are removed from the pypeit file. Here is a suggestion, but feel free to use a better approach:
```suggestion
    # Initialize setup
    ps0 = PypeItSetup.from_file_root(os.path.join(pargs.full_rawpath, pargs.root), script_Utils.spectrograph.name, extension='.fits')
    # run it
    ps0.run(setup_only=True)
    # no science file list
    file_list = np.array(ps0.file_list)[ps0.fitstbl['frametype'] != 'science'].tolist()
    # Initialize setup without science files
    ps = PypeItSetup(file_list=file_list, spectrograph_name=ps0.spectrograph.name)
    ps.run(setup_only=True)
    setups, indx = ps.fitstbl.get_configuration_names(return_index=True)
```

Reply:
Gack, I modified `fitstbl` but should have only modified `usrdata`.

Have put the change in and have confirmed we reduce one and only one `science` frame.
Thanks for catching this (again!).

I've also added a "kludge" for backwards compatibility in `util.py:_parse_setup_lines()`
not elegant but successful.  Re-starting the full DevSuite